src/libsystemd/sd-bus/bus-socket.c

```diff
@@ -156,93 +156,113 @@ static int bus_socket_write_auth(sd_bus *b) {
 }
 
 static int bus_socket_auth_verify_client(sd_bus *b) {
-        char *d, *e, *f, *start;
+        char *l, *lines[4] = {};
         sd_id128_t peer;
+        size_t i, n;
         int r;
 
         assert(b);
 
         /*
-         * We expect three response lines:
-         *   "DATA\r\n"
+         * We expect up to three response lines:
+         *   "DATA\r\n"                 (optional)
          *   "OK <server-id>\r\n"
          *   "AGREE_UNIX_FD\r\n"        (optional)
          */
 
-        d = memmem_safe(b->rbuffer, b->rbuffer_size, "\r\n", 2);
-        if (!d)
-                return 0;
-
-        e = memmem_safe(d + 2, b->rbuffer_size - (d - (char*) b->rbuffer) - 2, "\r\n", 2);
-        if (!e)
-                return 0;
-
-        if (b->accept_fd) {
-                f = memmem_safe(e + 2, b->rbuffer_size - (e - (char*) b->rbuffer) - 2, "\r\n", 2);
-                if (!f)
-                        return 0;
+        n = 0;
+        lines[n] = b->rbuffer;
+        l = memmem_safe(b->rbuffer, b->rbuffer_size, "\r\n", 2);
+        if (l) {
+                lines[++n] = l + 2;
+                l = memmem_safe(l + 2, b->rbuffer_size - (l - (char*) b->rbuffer) - 2, "\r\n", 2);
+                if (l) {
+                        lines[++n] = l + 2;
+                        l = memmem_safe(l + 2, b->rbuffer_size - (l - (char*) b->rbuffer) - 2, "\r\n", 2);
+                        if (l)
+                                lines[++n] = l + 2;
+                }
+        }
 
-                start = f + 2;
-        } else {
-                f = NULL;
-                start = e + 2;
+        /*
+         * If we sent a non-empty initial response, then we just expect an OK
```

Comment:
doesn't matter, but we usually line break at 109ch (.editorconfig, .vimrc, .dir-locals.el, .clang-format all carry that info btw)